src/_blog/metrics-in-go-libp2p.md

```diff
@@ -0,0 +1,182 @@
+
+```go
+
+func main() {
+        http.Handle("/metrics", promhttp.Handler())
+	go func() {
+		http.ListenAndServe(":2112", nil) // Any port is fine
+	}()
+
+	host, err := libp2p.New()
+        // err handling
+        ...
+}
+```
+Now just point your Prometheus instance to scrape from `:2122/metrics`
+
+By default, metrics are sent to the default Prometheus Registerer. To use a different Registerer from the default Prometheus registerer, use the option `libp2p.PrometheusRegisterer`. 
+
+```go
+
+func main() {
+	reg := prometheus.NewRegistry()
+        http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
+	go func() {
+		http.ListenAndServe(":2112", nil) // Any port is fine
+	}()
+
+	host, err := libp2p.New(
+                libp2p.PrometheusRegisterer(reg),
+        )
+        // err handling
+        ...
+}
+```
+ 
+### Discovering which Metrics are available
+
+go-libp2p provides metrics and Grafana dashboards for all its major subsystems out of the box. You can check <https://github.com/libp2p/go-libp2p/tree/master/dashboards> for the Grafana dashboards available. Another great way to discover available metrics is to open Prometheus ui and type `libp2p_(libp2p-package-name)_` and find available metrics from autocomplete. For Ex: `libp2p_autonat_` gives you the list of all metrics exported from [AutoNAT](https://github.com/libp2p/specs/tree/master/autonat).
+
+<div class="container" style="display:flex; column-gap:10px; justify-content: center; align-items: center;">
+    <figure>
+        <img src="../assets/metrics-in-go-libp2p-prometheus-ui.png" width="750">
+        <figcaption style="font-size:x-small;">
+                EvtLocalAddressesUpdated
+        </figcaption>
+    </figure>
+</div>
+
+To see the dashboards in action check the [Metrics and Dashboards](https://github.com/libp2p/go-libp2p/tree/master/examples/metrics-and-dashboards) example in the go-libp2p repo. This example sets up a dummy libp2p app configured with a Prometheus and Grafana instance. You can check all the dashboards available at <http://localhost:3000/dashboards>.
+
+### Local development and debugging setup
+
+We've made it extremely easy to get started with metrics for local development. You can use the docker setup provided in <https://github.com/libp2p/go-libp2p/tree/master/dashboards> to spin up a Grafana and Prometheus instance configured with all the available dashboards.
+
+First add these lines to your code. This exposes a metrics collection endpoint at <http://localhost:5001/debug/metrics/prometheus>
+
+```
+import "github.com/prometheus/client_golang/prometheus/promhttp"
+
+go func() {
+    http.Handle("/debug/metrics/prometheus", promhttp.Handler())
+    log.Fatal(http.ListenAndServe(":5001", nil))
+}()
+```
+
+Now run `docker compose up` and access your application's metrics at <http://localhost:3000>
```

Comment:
```suggestion
Now run `docker-compose up` and access your application's metrics at <http://localhost:3000>
```

This would need to be changed with https://github.com/libp2p/go-libp2p/pull/2397 (see README in that PR).

Reply:
`docker compose` is preferred over `docker-compose` I believe. When you run docker-compose it's actually just an alias for `docker compose`.

```
 docker-compose                        

Usage:  docker compose [OPTIONS] COMMAND
...
```

Reply:
I think `docker compose up` is the more popular variant. 
@marten-seemann no strong opinions here, I'll change if you want to.

Reply:
Looks like I'm not up-to-date with what's going on in the Docker world.